backend/api/professor/grading_materials.py

```
@@ -0,0 +1,333 @@
+                status_code=400,
+                detail=f"File {file.filename} too large. Maximum size is 10MB"
+            )
+        
+        # Extract text content based on file type
+        text_content = await _extract_text_content(content, file_extension)
+        
+        if not text_content or len(text_content.strip()) < 50:
+            raise HTTPException(
+                status_code=400,
+                detail=f"File {file.filename} contains insufficient text content"
+            )
```

Ruff (0.14.1)

Abstract `raise` to an inner function

(TRY301)